inbm/dispatcher-agent/tests/unit/sota/test_os_factory.py

```diff
@@ -33,15 +33,15 @@ def test_raise_error_unsupported_OsFactory(self) -> None:
         self.assertRaises(ValueError, factory.get_os, "MacOS")
 
     def test_create_ubuntu_snapshot_checker(self) -> None:
-        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu').create_snapshotter('update', '1', False)) \
+        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('Ubuntu').create_snapshotter('update', '1', False, 'Y')) \
             is DebianBasedSnapshot
 
     def test_create_yocto_snapshot_checker(self) -> None:
-        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('YoctoX86_64').create_snapshotter('update', '1', False)) \
+        assert type(SotaOsFactory(self.mock_disp_broker, None, []).get_os('YoctoX86_64').create_snapshotter('update', '1', False, 'Y')) \
             is YoctoSnapshot
```

Comment:
**Suggestion:** Replace assert statements with unittest's assertIsInstance for better test messages and consistency. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        yocto_os = SotaOsFactory(self.mock_disp_broker, None, []).get_os('YoctoX86_64')
        snapshotter = yocto_os.create_snapshotter('update', '1', False, 'Y')
        self.assertIsInstance(snapshotter, YoctoSnapshot)
```